providers-sdk/v1/vault/config.go

```diff
@@ -9,19 +9,32 @@ import (
 	"strings"
 )
 
+// supports both the protobuf string and the custom defined string
+// representations of a vault type
 func NewVaultType(name string) (VaultType, error) {
 	entry := strings.TrimSpace(strings.ToLower(name))
 	var code VaultType
-	ok := false
+	marshalMapOk := false
 	for k := range vaultMarshalNameMap {
 		if vaultMarshalNameMap[k] == entry {
-			ok = true
+			marshalMapOk = true
 			code = k
 			break
 		}
 	}
-	if !ok {
-		return VaultType_None, errors.New("unknown type value: " + string(name))
+	if !marshalMapOk {
```

Comment:
Do we then still need vaultMarshalNameMap? Or can we simplify the code?

Reply:
Looks like we're still using this in the `vault`cmds, e.g.  `vault list`

I think we should leave it to support inventory files where the type was defined with this custom map